packages/flutter/lib/src/foundation/internal.dart

```diff
@@ -0,0 +1,10 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// The largest SMI value.
+///
+/// See <https://www.dartlang.org/articles/numeric-computation/#smis-and-mints>
+///
+/// When compiled to JavaScript, this value is intead the largest 32bit int.
```

Comment:
Is this ok? Wondering what the implications are for just taking your app and re-compiling it for the web?

Reply:
Long term we should deprecate and remove this class, but generally I think changing behavior is inescapable (FileImage, et cetera). The alternative is two separate programs.